### ISSUE TITLE
Where to put custom DSP plots?

### ISSUE THREAD

**Issue by standarddeviant** [JuliaPlots/StatPlots.jl#147]
Hello,

I apologize if this isn't the right place for this discussion.  I would like to get some plot functions/recipes in to a package that is aware of custom types in DSP.jl, there's a discussion here:
https://github.com/JuliaDSP/DSP.jl/issues/203 

I promised to follow up with StatPlots.jl devs on a question from that discussion:
Do people think StatPlots.jl would be a good home for DSP.jl-aware plots?

**Comment by piever** [comment]
DSP seems lightweight enough that adding a dependency here shouldn't be a big deal. I'm in favor to adding here a dependency to have a recipe to plot objects from signal processing.
@mkborregaard , what do you think?

**Comment by mkborregaard** [comment]
I agree with the discussion on the issue that a more relevant place to put it would be in DSP itself, but failing that it'd be fine to have it here or in PlotRecipes. I don't think it will interact with any functionality here in StatPlots which might be an argument for PlotRecipes, but on the other hand that's a dependency-heavy package.

**Comment by ssfrr** [comment]
I think StatsPlots would be a weird place for this functionality - it seems unrelated to all things statistical. I guess PlotRecipes would be reasonable, but then you'd need to add a dependency to `DSP` to it, which doesn't seem like a scalable way to do things (if people keep doing that, PlotRecipes ends up being this sprawling octopus package with dependencies on all sorts of unrelated package).

**Comment by mkborregaard** [comment]
I agree with that completely. Adding this recipe to DSP itself is by far the most sustainable approach, and in fact it comes with no disadvantages to anyone. However, some developers on the statistical packages disagree with adding plot recipes on principle; discussions like the one on that issue has been had more times than I can count on these packages. StatPlots and PlotRecipes represent a compromise for still making recipe functionality available and discoverable to the community, with exactly the consequence you describe, at least for PlotRecipes. Now we've decided that StatPlots should depend mainly on the same packages as Stats.jl that will be more coherent.

Any good suggestions for resolving this are greatly appreciated.

**Comment by mkborregaard** [comment]
Did you put this in DSP @standarddeviant ? I see the linked issue is closed
